Makes `styled-components` a peerDependency 💅

fiber-god opened the pull request:
Hello,

This PR makes `styled-components` a peer-dep. This is my first Status Open Bounty for me which is also pretty exciting. 

Happy OSS!

-- Juwan 

Fixes #35

sohkai commented:
Hey Juwan (@fiber-god)!

Awesome, thanks for doing this! We'll fund the bounty tomorrow; hope you don't mind waiting a bit :).

It also might take us a bit of time to merge this in, since we have to make sure everything works correctly with our dependent projects (e.g. [aragon/aragon](https://github.com/aragon/aragon). We've been hesitating on doing this ourselves because we're not sure if everything will just work out of the box (fingers crossed).

fiber-god commented:
I've removed the re-export, and added styled-components as a dependency for the gallery. I can move gallery's react, react-dom, and styled-components dep to peerdependencies too if needed, but testing how it works on the root project should probably be done first 👍

sohkai commented on .gitignore:
```diff
@@ -2,3 +2,4 @@ dist/
 lib/
 node_modules/
 npm-debug.log
+.vscode/
```
Do you mind reverting this change? Editor-specific config should usually go into your global .gitignore :).

fiber-god replied:
Yep! I actually didn't know that global .gitignore's were a thing, so thanks for the insight!

fiber-god commented:
Those extra commits were generated from running a rebase on my current PR. I ran:

```
❯ git remote add upstream https://github.com/aragon/aragon-ui.git
❯ git fetch upstream
❯ git rebase upstream/master
```

It shouldn't affect my changes (it only keeps it in sync with Aragon's master), but I just wanted to explain why these commits appeared.

sohkai commented on src/components/SidePanel/SidePanel.js:
```diff
@@ -62,9 +62,12 @@ const StyledPanelCloseButton = styled.button`
       border: 0;
     }
+
+    ${unselectable()};
```
Hmm, not sure why these changes are showing up here if you rebased them. Can you try pulling and rebasing again?

fiber-god replied:
just rebased and those changes don't seem to be in the latest push

sohkai replied:
Hmm, now it's showing the newest SidePanel changes. Really weird...

fiber-god replied:
That's how it should be right? Rebase replays my work on top of the newest `master` changes

sohkai replied:
Well, you're right, but the diff of the SidePanel changes shouldn't be showing up on Github. Not sure what's going on; my local git diff doesn't show it as being different.

Anyhow, don't worry about it :).

sohkai commented:
644e1059f8c65149470ca3214dc1a74142f69722 seems like it should probably be reverted?

sohkai commented:
Going to merge since it seems to be working seamlessly with https://github.com/aragon/aragon/pull/91 :).

Thanks @fiber-god for taking the effort to do this!
